src/index.js

```diff
@@ -337,15 +337,22 @@ Penpal.connectToParent = ({ parentOrigin, methods = {} }) => {
 
   const child = window;
   const parent = child.parent;
+  const targetParentOrigin = getOriginFromUrl(document.referrer);
 
   if (parentOrigin !== undefined && !Array.isArray(parentOrigin)) {
     parentOrigin = [parentOrigin];
   }
 
+  if (parentOrigin !== undefined && parentOrigin.indexOf(targetParentOrigin) === -1) {
+    log('Child: actual parent origin not in list of valid parent origins, Penpal connection not created');
```

Comment:
On second thought, I think it would be better if this just threw an error. Logging is intended to provide info about what's going on, but if we know something fails, it's probably best to throw an error. I'll change this after merging.